crates/tracing/src/lib.rs

```diff
@@ -114,18 +114,21 @@ pub fn configure_tracing(
 
 #[cfg(feature = "otel")]
 #[allow(clippy::missing_errors_doc)] // TODO: Document errors
+#[allow(clippy::needless_pass_by_value)]
 pub fn configure_tracing(
     service_name: String,
     otel_config: &OtelConfig,
     structured_logging_enabled: bool,
     log_level_override: Option<&Level>,
 ) -> anyhow::Result<()> {
+    use heck::ToKebabCase;
```

Comment:
Nit: unless there's a good reason to use inline imports (e.g. purposefully limiting scope to avoid a naming conflict, or conditional compilation), we prefer importing once at the top of the file. Makes things easier to find

Reply:
Fair enough, I chose to import it here because this was the only place it was being used, so it would be local to the code that consumes it. I'm happy to hoist it up to the top if that's more appropriate.

Reply:
I don't have a strong opinion, but curious how @brooksmtownsend @thomastaylor312 feel

most importantly I wish we had automated tooling that just formatted these things for us :)

Reply:
I went ahead and hoisted the use statement up to the top 👍

Reply:
For future reference so my opinion is out in public 😆, I am really not a fan of code that imports in a function unless for the reasons @connorsmith256. I find that they clutter code and make it hard to see what is all actually imported